spec/support/matchers/flash_matchers.rb

```diff
@@ -0,0 +1,11 @@
+RSpec::Matchers.define :have_flash_message do |flash_type|
+  match do |page|
+    if flash_type.present?
+      page.has_css?("div.flash_#{flash_type}")
+      within("div.flashes") { page.has_text }
```

Comment:
Does `has_text`get passed something? Or does it assert that *any* text is present?

Can you achieve something similar in on shot via `page.has_css?("div.flash_#{flash_type}", text: options)`?